packages/core/src/Parser.ts

```diff
@@ -85,42 +84,57 @@ export class Parser {
             } while (context);
         }
 
-        // Parse the DOM range if any.
+        // Parse the DOM selection.
         const selection = node.ownerDocument.getSelection();
-        const range = selection.rangeCount && selection.getRangeAt(0);
-        if (range && node.contains(range.startContainer) && node.contains(range.endContainer)) {
-            const forward = range.startContainer === selection.anchorNode;
-            const vRange = this.parseRange(range, forward ? Direction.FORWARD : Direction.BACKWARD);
-            vDocument.range.set(vRange);
+        if (
+            selection &&
+            node.contains(selection.anchorNode) &&
+            node.contains(selection.focusNode)
+        ) {
+            vDocument.selection.set(this.parseSelection(selection));
         }
 
-        // Set a default range in VDocument if none was set yet.
-        if (!vDocument.range.start.parent || !vDocument.range.end.parent) {
-            vDocument.range.setAt(vDocument.root);
+        // Set a default selection in VDocument if none was set yet.
+        if (!vDocument.selection.anchor.parent || !vDocument.selection.focus.parent) {
+            vDocument.selection.setAt(vDocument.root);
         }
 
         return vDocument;
     }
     /**
-     * Convert the DOM description of a range to the description of a VRange.
+     * Parse the dom selection into the description of a VSelection.
      *
-     * @param range
+     * @param selection
      * @param [direction]
      */
-    parseRange(range: DomRangeDescription): VRangeDescription;
-    parseRange(range: Range, direction: Direction): VRangeDescription;
-    parseRange(range: Range | DomRangeDescription, direction?: Direction): VRangeDescription {
-        const start = this._locate(range.startContainer, range.startOffset);
-        const end = this._locate(range.endContainer, range.endOffset);
+    parseSelection(selection: Selection | DomSelectionDescription): VSelectionDescription {
```

Comment:
Well `Selection` is the true class of the DOM, while `DomSelectionDescription` is when we describe a selection by specifying an `anchorNode`, a `focusNode` etc. This method is called by the `EventManager` to convert the description sent by the `EventNormalizer` and it is called by the `Parser` itself, in which case it gives it the true `Selection` of the DOM. Neither type is a subtype of the other because `Selection` has a lot more than the 4 properties we use *but* it is missing `direction`. I think we could recompute `direction` here though so it could be possible to define an interface that would suit both cases. The `EventNormalizer` has to compute the direction anyway to give it to us here so it's not like we would add additional computional work by computing it here instead.

Reply:
I think it's the `Normalizer`'s job to make a `DomSelectionDescription` object from the native `Selection`: it's normalizing it precisely so our parser always receives the same type of object and not the native stuff.